qp_qiime2/__init__.py

```diff
@@ -76,234 +86,191 @@
-                    req_params[ename] = ('artifact', [etype])
-                # we need to add the actual name of the parameter so we
-                # can retrieve later
-                req_params['qp-hide-param' + ename] = ('string', pname)
-
-            for td in to_delete:
-                del inputs[td]
+    # storing this information in req_params so we can use internally
+    # while calling call_qiime2
+    req_params = {'qp-hide-plugin': ('string', q2plugin.name),
+                  'qp-hide-method': ('string', m.id)}
+    outputs_params = {}
+    opt_params = {}
+    to_delete = []
+    for pname, element in inputs.items():
+        qt_name, predicate = get_qiime2_type_name_and_predicate(element)
 
-            for pname, element in outputs.items():
-                qt_name, predicate = get_qiime2_type_name_and_predicate(
-                    element)
-                if qt_name not in Q2_QIITA_SEMANTIC_TYPE:
-                    add_method = False
-                    break
-                else:
-                    etype = Q2_QIITA_SEMANTIC_TYPE[qt_name]
-                    outputs_params[pname] = etype
+        if qt_name not in Q2_QIITA_SEMANTIC_TYPE:
+            add_method = False
+            break
 
-            if len(inputs) != 1 or not add_method:
-                # This is currently filtering out:
-                # diversity mantel
-                # diversity pcoa_biplot
-                # diversity pcoa_biplot
-                # diversity procrustes_analysis
-                # emperor procrustes_plot
-                # gneiss assign_ids
-                # gneiss assign_ids
-                # gneiss balance_taxonomy
-                # gneiss balance_taxonomy
-                # gneiss correlation_clustering
-                # gneiss dendrogram_heatmap
-                # gneiss gradient_clustering
-                # gneiss gradient_clustering
-                # gneiss ilr_hierarchical
-                # gneiss ilr_phylogenetic
-                # gneiss ilr_phylogenetic
-                # longitudinal feature_volatility
-                # longitudinal maturity_index
-                # sample-classifier classify_samples
-                # sample-classifier fit_classifier
-                # sample-classifier fit_regressor
-                # sample-classifier predict_classification
-                # sample-classifier predict_regression
-                # sample-classifier regress_samples
-                # taxa filter_seqs
-                continue
+        add_method = True
+        etype = Q2_QIITA_SEMANTIC_TYPE[qt_name]
+        # these are special types as we can retrive internally
+        if etype in 'phylogeny':
+            ename = 'Phylogenetic tree'
+            req_params[ename] = (
+                'choice:["None", "Artifact tree, if exists"]', 'None')
+            req_params['qp-hide-param' + ename] = ('string', pname)
+            to_delete.append(pname)
+        elif etype == 'FeatureData':
```

Comment:
These is the only "real" change to support the change in https://github.com/qiita-spots/qtp-diversity/pull/13 and allow the `predicate == ['Sequence']` block.